### PR TITLE
Feat/next/ber

### PR DESCRIPTION
**Pull request by gundelsby**

### CODE REVIEW

File: Ber/Reader.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    let exponent = 0;
+
+    if(buf.readUInt8(o) & 0x80) {
+      exponent = -1;
+    }
+
+    if(buf.length - o < exponentLength) {
+      throw new errors.ASN1Error('Invalid ASN.1; not enough length to contain exponent');
+    }
+
```
Comment (mint-dewit) [inline Ber/Reader.js:81]: What happened to ber.js line 157?

File: Ber/Reader.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    let significand = new Long(0, 0, true)
+    while(o < buf.length) {
+      significand = significand.shl(8).or(buf.readUInt8(o++))
+    }
+
+    significand = significand.shl(significandShift)
+
```
Comment (mint-dewit) [inline Ber/Reader.js:92]: Missing ber.js line 173?

**Review by mint-dewit — CHANGES_REQUESTED** [review]
It appears this PR breaks this test case: https://github.com/nrkno/tv-automation-emberplus-connection/blob/next/test/Ember.test.js#L504 (Could very well be that this will be resolved by adding the missing lines from ber.js?)

File: Ber/Reader.js (hunk 1 above)
Comment (gundelsby) [reply Ber/Reader.js:81]: You will find that at Reader.js:82:
`    let significand = new Long(0, 0, true)`

I moved it down to right over the first statement that actually uses the variable.

File: Ber/Reader.js (hunk 2 above)
Comment (gundelsby) [reply Ber/Reader.js:92]: Well, that line is empty in the original ber.js? What statement are you thinking about?

**Comment by gundelsby** [comment]
> 
> 
> It appears this PR breaks this test case: https://github.com/nrkno/tv-automation-emberplus-connection/blob/next/test/Ember.test.js#L504 (Could very well be that this will be resolved by adding the missing lines from ber.js?)

Yup, that's a test that passes in the next branch. I'll try to figure it out :)

File: Ber/Reader.js (hunk 2 above)
Comment (mint-dewit) [reply Ber/Reader.js:92]: I think you may be looking at the master branch, I'm looking at the next branch: https://github.com/nrkno/tv-automation-emberplus-connection/blob/next/ber.js#L173

File: Ber/Reader.js (hunk 1 above)
Comment (mint-dewit) [reply Ber/Reader.js:81]: I meant this loop actually https://github.com/nrkno/tv-automation-emberplus-connection/blob/next/ber.js#L157

File: Ber/Reader.js (hunk 2 above)
Comment (gundelsby) [reply Ber/Reader.js:92]: 🤦‍♂️

**Review by mint-dewit — APPROVED** [review]

**Pull request merged**